### PR TITLE
Handle breaking changes

### PR DESCRIPTION
**Pull request by FrederikBolding**
Handle breaking changes with the release of Flask `10.25.0`. 

- Update dependencies
- Update changed parameters for RPC methods
- Leverage custom UI for the example snap
- Add `endowment:rpc`

Fixes https://github.com/MetaMask/template-snap-monorepo/issues/23

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,6 +79,6 @@ The project follows the same release process as the other libraries in the MetaM
 - For the global `wallet` type to work, you have to add the following to your `tsconfig.json`:
   ```json
   {
-    "files": ["./node_modules/@metamask/snap-types/global.d.ts"]
+    "files": ["./node_modules/@metamask/snaps-types/global.d.ts"]
   }
   ```
```
Comment (Mrtenz) [inline README.md:84]: This is no longer necessary. The global types are automatically imported as long as the snap is using an import from `snaps-types`.

File: packages/site/tsconfig.json [hunk 2]
```diff
@@ -7,6 +7,6 @@
     "noEmit": true,
     "jsx": "preserve"
   },
-  "files": ["../../node_modules/@metamask/snap-types/global.d.ts"],
+  "files": ["../../node_modules/@metamask/snaps-types/global.d.ts"],
```
Comment (Mrtenz) [inline packages/site/tsconfig.json:10]:
```suggestion
```

File: packages/snap/README.md [hunk 3]
```diff
@@ -9,6 +9,6 @@ This Snap demonstrates how to develop a Snap with TypeScript.
 - For the global `wallet` type to work, you have to add the following to your `tsconfig.json`:
   ```json
   {
-    "files": ["./node_modules/@metamask/snap-types/global.d.ts"]
+    "files": ["./node_modules/@metamask/snaps-types/global.d.ts"]
   }
   ```
```
Comment (Mrtenz) [inline packages/snap/README.md:14]: Same here as above.

File: packages/snap/src/index.ts [hunk 4]
```diff
@@ -16,24 +8,24 @@ export const getMessage = (originString: string): string =>
...
-        params: [
-          {
-            prompt: getMessage(origin),
-            description:
-              'This custom confirmation is just for display purposes.',
-            textAreaContent:
+      return snap.request({
+        method: 'snap_dialog',
+        params: {
+          type: 'Confirmation',
```
Comment (Mrtenz) [inline packages/snap/src/index.ts:20]:
```suggestion
          type: DialogType.Confirmation,
```

File: packages/snap/tsconfig.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
   "extends": "../../tsconfig.json",
-  "files": ["../../node_modules/@metamask/snap-types/global.d.ts"],
+  "files": ["../../node_modules/@metamask/snaps-types/global.d.ts"],
```
Comment (Mrtenz) [inline packages/snap/tsconfig.json:3]:
```suggestion
```

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (FrederikBolding) [reply README.md:84]: Is that true for `0.27.1`?

Comment (Mrtenz) [reply README.md:84]: Yes. It's part of [this commit](https://github.com/MetaMask/snaps-monorepo/commit/b3ce1ffdf7bd5ed125d53f3fd489d3adfacfb9fb), which was first released in 0.27.0.

Is there any reason for us not to use the latest version, though?

Comment (FrederikBolding) [reply README.md:84]: Ah sweet, I didn't realize that. I don't want to use the latest version as there may be breaking changes not part of the `10.25.0` release.

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**